dev/com.ibm.ws.jdbc/src/com/ibm/ws/jdbc/internal/JDBCDriverService.java

```diff
@@ -350,21 +350,29 @@ public CommonDataSource createAnyDataSource(Properties props) throws SQLExceptio
             String vendorPropertiesPID = props instanceof PropertyService ? ((PropertyService) props).getFactoryPID() : PropertyService.FACTORY_PID;
             String className;
 
+            // First check DataSource impl class names based on direct user configuration or db-specific properties elements 
```

Comment:
Cannot do this, it would be a breaking change to anyone who specifies just one of the data source impl classes on a jdbcDriver, where they have multiple data sources, some of which use the type for which the impl is specified, and others of which use the other types and expect to get the inferred default impls for those types.
A better approach (which should probably be a separate issue) is to trap the ClassNotFoundException, and only on the exception path, continue to try the other types, in the documented order, until one is found.